.github/workflows/unit-test-workflow.yml

```diff
@@ -0,0 +1,36 @@
+name: Unit test workflow
+
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python: ["3.9", "3.10", "3.11"]
```

Comment:
What's the reason to run the matrix on all three of these? Just to make sure you're writing forward-compatible code? The lambda in theory should run on exactly one version, the CLI I suppose should be tested against all currently supported versions.

Reply:
this was how refreezer was setup and I saw no reason to remove it. Generally yes, I think the idea is to just make sure there are no weird breakages between versions to avoid being caught out in the long-term